### PR TITLE
Add table aws_cloudformation_stack_set Closes #1811

### LINKED ISSUE #1811: Add table aws_cloudformation_stack_set
AWS CloudFormation StackSets are CloudFormation Stacks that can be deployed from one to many accounts.

They are a different type of object than CloudFormation Stacks on the "deployer" account. 

Reference: https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_ListStackSets.html

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```

No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_cloudformation_stack_set []

PRETEST: tests/aws_cloudformation_stack_set

TEST: tests/aws_cloudformation_stack_set
Running terraform
data.aws_region.alternate: Reading...
data.aws_region.alternate: Read complete after 0s [id=us-east-2]
data.aws_region.primary: Reading...
data.aws_caller_identity.current: Reading...
data.aws_partition.current: Reading...
data.aws_iam_policy_document.AWSCloudFormationStackSetAdministrationRole_assume_role_policy: Reading...
data.aws_partition.current: Read complete after 0s [id=aws]
data.aws_region.primary: Read complete after 0s [id=us-east-1]
data.aws_iam_policy_document.AWSCloudFormationStackSetAdministrationRole_assume_role_policy: Read complete after 0s [id=478182822]
data.aws_caller_identity.current: Read complete after 1s [id=222222222222]
data.null_data_source.resource: Reading...
data.null_data_source.resource: Read complete after 0s [id=static]

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create
 <= read (data resources)

Terraform will perform the following actions:

  # data.aws_iam_policy_document.AWSCloudFormationStackSetAdministrationRole_ExecutionPolicy will be read during apply
  # (config refers to values not yet known)
 <= data "aws_iam_policy_document" "AWSCloudFormationStackSetAdministrationRole_ExecutionPolicy" {
      + id   = (known after apply)
      + json = (known after apply)

      + statement {
          + actions   = [
              + "sts:AssumeRole",
            ]
          + effect    = "Allow"
          + resources = [
              + (known after apply),
            ]
        }
    }

  # aws_cloudformation_stack_set.example will be created
  + resource "aws_cloudformation_stack_set" "example" {
      + administration_role_arn = (known after apply)
      + arn                     = (known after apply)
      + call_as                 = "SELF"
      + execution_role_name     = (known after apply)
      + id                      = (known after apply)
      + name                    = "turbottest92043"
      + parameters              = {
          + "VPCCidr" = "10.0.0.0/16"
        }
      + permission_model        = "SELF_MANAGED"
      + stack_set_id            = (known after apply)
      + tags_all                = (known after apply)
      + template_body           = jsonencode(
            {
              + Parameters = {
                  + VPCCidr = {
                      + Default     = "10.0.0.0/16"
                      + Description = "Enter the CIDR block for the VPC. Default is 10.0.0.0/16."
                      + Type        = "String"
                    }
                }
              + Resources  = {
                  + myVpc = {
                      + Properties = {
                          + CidrBlock = {
                              + Ref = "VPCCidr"
                            }
                          + Tags      = [
                              + {
                                  + Key   = "Name"
                                  + Value = "Primary_CF_VPC"
                                },
                            ]
                        }
                      + Type       = "AWS::EC2::VPC"
                    }
                }
            }
        )
    }

  # aws_iam_role.AWSCloudFormationStackSetAdministrationRole will be created
  + resource "aws_iam_role" "AWSCloudFormationStackSetAdministrationRole" {
      + arn                   = (known after apply)
      + assume_role_policy    = jsonencode(
            {
              + Statement = [
                  + {
                      + Action    = "sts:AssumeRole"
                      + Effect    = "Allow"
                      + Principal = {
                          + Service = "cloudformation.amazonaws.com"
                        }
                    },
                ]
              + Version   = "2012-10-17"
            }
        )
      + create_date           = (known after apply)
      + force_detach_policies = false
      + id                    = (known after apply)
      + managed_policy_arns   = (known after apply)
      + max_session_duration  = 3600
      + name                  = "AWSCloudFormationStackSetAdministrationRole"
      + name_prefix           = (known after apply)
      + path                  = "/"
      + tags_all              = (known after apply)
      + unique_id             = (known after apply)
    }

  # aws_iam_role_policy.AWSCloudFormationStackSetAdministrationRole_ExecutionPolicy will be created
  + resource "aws_iam_role_policy" "AWSCloudFormationStackSetAdministrationRole_ExecutionPolicy" {
      + id     = (known after apply)
      + name   = "ExecutionPolicy"
      + policy = (known after apply)
      + role   = "AWSCloudFormationStackSetAdministrationRole"
    }

Plan: 3 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + account_id    = "222222222222"
  + region_name   = "us-east-1"
  + resource_aka  = (known after apply)
  + resource_id   = (known after apply)
  + resource_name = "turbottest92043"
aws_iam_role.AWSCloudFormationStackSetAdministrationRole: Creating...
aws_iam_role.AWSCloudFormationStackSetAdministrationRole: Creation complete after 3s [id=AWSCloudFormationStackSetAdministrationRole]
aws_cloudformation_stack_set.example: Creating...
aws_cloudformation_stack_set.example: Creation complete after 3s [id=turbottest92043]
data.aws_iam_policy_document.AWSCloudFormationStackSetAdministrationRole_ExecutionPolicy: Reading...
data.aws_iam_policy_document.AWSCloudFormationStackSetAdministrationRole_ExecutionPolicy: Read complete after 0s [id=3361755086]
aws_iam_role_policy.AWSCloudFormationStackSetAdministrationRole_ExecutionPolicy: Creating...
aws_iam_role_policy.AWSCloudFormationStackSetAdministrationRole_ExecutionPolicy: Creation complete after 0s [id=AWSCloudFormationStackSetAdministrationRole:ExecutionPolicy]

Warning: Deprecated

  with data.null_data_source.resource,
  on variables.tf line 44, in data "null_data_source" "resource":
  44: data "null_data_source" "resource" {

The null_data_source was historically used to construct intermediate values
to re-use elsewhere in configuration, the same can now be achieved using
locals

(and one more similar warning elsewhere)

Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

account_id = "222222222222"
region_name = "us-east-1"
resource_aka = "arn:aws:cloudformation:us-east-1:222222222222:stackset/turbottest92043:1409f020-6d2d-463b-8ec3-23d5e58ae66b"
resource_id = "turbottest92043:1409f020-6d2d-463b-8ec3-23d5e58ae66b"
resource_name = "turbottest92043"

Running SQL query: query.sql
[
  {
    "arn": "arn:aws:cloudformation:us-east-1:222222222222:stackset/turbottest92043:1409f020-6d2d-463b-8ec3-23d5e58ae66b",
    "stack_set_id": "turbottest92043:1409f020-6d2d-463b-8ec3-23d5e58ae66b",
    "stack_set_name": "turbottest92043"
  }
]
✔ PASSED

Running SQL query: test-get-query.sql
[
  {
    "stack_set_id": "turbottest92043:1409f020-6d2d-463b-8ec3-23d5e58ae66b",
    "stack_set_name": "turbottest92043",
    "status": "ACTIVE",
    "title": "turbottest92043"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "stack_set_id": "turbottest92043:1409f020-6d2d-463b-8ec3-23d5e58ae66b",
    "stack_set_name": "turbottest92043"
  }
]
✔ PASSED

POSTTEST: tests/aws_cloudformation_stack_set

TEARDOWN: tests/aws_cloudformation_stack_set

SUMMARY:

1/1 passed.


```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select * from aws_cloudformation_stack_set
+---------------------------------------------------------------------------+--------------------------------------+----------------------------------------------------------------------------------------------------------------------------------+---------+------------>
| stack_set_id                                                              | stack_set_name                       | arn                                                                                                                              | status  | description>
+---------------------------------------------------------------------------+--------------------------------------+----------------------------------------------------------------------------------------------------------------------------------+---------+------------>
| turbottest92043:1409f020-6d2d-463b-8ec3-23d5e58ae66b                      | turbottest92043                      | <null>                                                                                                                           | DELETED | <null>     >
| turbottest15563:dc284814-48ef-46e4-ad38-b2348345797e                      | turbottest15563                      | <null>                                                                                                                           | DELETED | <null>     >
| turbottest17944:f21726d9-be19-4e14-b7b2-7da52066658c                      | turbottest17944                      | <null>                                                                                                                           | DELETED | <null>     >
| turbottest44407:35f394e4-3de8-48be-bd69-1effe6ee2869                      | turbottest44407                      | <null>                                                                                                                           | DELETED | <null>     >
| AWS-QuickSetup-CFGRecording-LA-v9hwh:7be49155-cb25-4c02-b287-352006ed6c6f | AWS-QuickSetup-CFGRecording-LA-v9hwh | arn:aws:cloudformation:us-east-1:222222222222:stackset/AWS-QuickSetup-CFGRecording-LA-v9hwh:7be49155-cb25-4c02-b287-352006ed6c6f | ACTIVE  | Enables the>
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
|                                                                           |                                      |                                                                                                                                  |         |            >
+---------------------------------------------------------------------------+--------------------------------------+----------------------------------------------------------------------------------------------------------------------------------+---------+------------>

Time: 7.6s. Rows fetched: 5. Hydrate calls: 10.

> select * from aws_cloudformation_stack_set where stack_set_name = 'AWS-QuickSetup-CFGRecording-LA-v9hwh'
+---------------------------------------------------------------------------+--------------------------------------+----------------------------------------------------------------------------------------------------------------------------------+--------+------------->
| stack_set_id                                                              | stack_set_name                       | arn                                                                                                                              | status | description >
+---------------------------------------------------------------------------+--------------------------------------+----------------------------------------------------------------------------------------------------------------------------------+--------+------------->
| AWS-QuickSetup-CFGRecording-LA-v9hwh:7be49155-cb25-4c02-b287-352006ed6c6f | AWS-QuickSetup-CFGRecording-LA-v9hwh | arn:aws:cloudformation:us-east-1:222222222222:stackset/AWS-QuickSetup-CFGRecording-LA-v9hwh:7be49155-cb25-4c02-b287-352006ed6c6f | ACTIVE | Enables the >
|                                                                           |                                      |                                                                                                                                  |        |             >
|                                                                           |                                      |                                                                                                                                  |        |             >
|                                                                           |                                      |                                                                                                                                  |        |             >
|                                                                           |                                      |                                                                                                                                  |        |             >
|                                                                           |                                      |                                                                                                                                  |        |            

```
</details>

### CODE REVIEW

File: aws/table_aws_cloudformation_stack_set.go [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+//// TABLE DEFINITION
+
+func tableAwsCloudFormationStackSet(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_cloudformation_stack_set",
+		Description: "AWS CloudFormation Stack Set",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("stack_set_name"),
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"ResourceNotFoundException", "StackSetNotFoundException"}),
```
Comment (misraved) [inline aws/table_aws_cloudformation_stack_set.go:26]: @ParthaI Do we need to ignore both the error codes?

**Review by misraved — COMMENTED** [review]
@ParthaI please take a look at the review comments. Thanks!!

File: aws/table_aws_cloudformation_stack_set.go [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+func getCloudFormationStackSet(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	var name string
+
+	if h.Item != nil {
+		data := h.Item.(types.StackSetSummary)
+		name = *data.StackSetName
+	} else {
+		name = d.EqualsQuals["stack_set_name"].GetStringValue()
```
Comment (misraved) [inline aws/table_aws_cloudformation_stack_set.go:234]:
```suggestion
		name = d.EqualsQuals["stack_set_name"].GetStringValue()
```
@ParthaI can we use `EqualsQualString` instead?

File: aws/table_aws_cloudformation_stack_set.go (hunk 1 above)
Comment (ParthaI) [reply aws/table_aws_cloudformation_stack_set.go:26]: @misraved, Apologies for the confusion. We actually only need to ignore the error code `StackSetNotFoundException`. According to the [documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_DescribeStackSet.html), it states that if the resource does not exist, we should receive a `StackSetNotFound` error. However, in our case, we are receiving a `StackSetNotFoundException` error instead.

File: aws/table_aws_cloudformation_stack_set.go (hunk 2 above)
Comment (ParthaI) [reply aws/table_aws_cloudformation_stack_set.go]: Updated

**Review by madhushreeray30 — APPROVED** [review]
LGTM 👍

**Pull request merged**